File: Domains/Base/Workspace/Autotest/Angular/src/Compiler/Route.ts
import { StaticSymbol } from '@angular/compiler';
import { ModuleSymbol } from 'ngast';

export interface Route {
  path?: string
  pathMatch?: string
  // matcher?: UrlMatcher
  component?: StaticSymbol // Type<any>
  redirectTo?: string
  outlet?: string
  canActivate?: any[]
  canActivateChild?: any[]
  canDeactivate?: any[]
  canLoad?: any[]
  data?: any // Data
  // resolve?: ResolveData
  children?: Route[]
  loadChildren?: string // LoadChildren
  // runGuardsAndResolvers?: RunGuardsAndResolvers

  module?: ModuleSymbol
}

File: Domains/Base/Workspace/Autotest/Model/Export/Base/Angular/Route.cs
using Newtonsoft.Json.Linq;

namespace Autotest.Angular
{
    public class Route
    {
    public string Path { get; set; }
    public string PathMatch{ get; set; }
    public Directive Component{ get; set; }
    public string RedirectTo{ get; set; }
    public string Outlet{ get; set; }
    public JToken Data{ get; set; }
    public Route[] Children{ get; set; }
}
}